### PR TITLE
Current

### PR DESCRIPTION
**Pull request by takeramagan**
add auth

### CODE REVIEW

File: src/hooks/useRequest.ts [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
-export const useRequest = () =>{
-  const [loading, setLoading] = useState(true)
+export const useRequest = (method: 'get'|'post' = 'get', initLoading: true | false = true) =>{
+  const [loading, setLoading] = useState(initLoading)
   const [error, setError] = useState(false)
   
-  const request = async <T>(endPoint: string, handler:(data :T) => void | T) => {
+  const request = async <T>(
+    endPoint: string, 
+    datahandler:(data :T) => void | T,
```
Comment (Yuanzjls) [inline src/hooks/useRequest.ts:11]: datahandler should be dataHandler

**Review by Yuanzjls — APPROVED** [review]
Awesome work. It is impressing to see the loading page when the program is fetching data from server. Beautiful file structures, which makes it easy to expand and maintain.

**Comment by takeramagan** [comment]
add users page

File: src/pages/EventDetailPage/Event.tsx [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
-import moment from 'moment';
+import moment, { now } from 'moment';
 import GoogleMapReact from 'google-map-react';
 import PinDropIcon from '@material-ui/icons/PinDrop';
+import { Review } from 'components';
+import { useRequest } from 'hooks';
+import { APP_END_POINT } from '../../constants'
+import jwt_decode from "jwt-decode";
+import { AddReview } from './AddReview';
+import { type } from 'node:os';
```
Comment (sammyluo) [inline src/pages/EventDetailPage/Event.tsx:15]: This can be removed

File: src/pages/EventDetailPage/Event.tsx [hunk 3]
```diff
@@ -22,9 +28,101 @@ const Marker: React.FC<MarkerProps> = ({text, lat, lng}) => {
...
+  const {token, addReview:addReviewToLocalList} = props
+  const onAttend = () =>{
+    if(loading) return //过滤请求中的多次点击
+    setShowAttendConfirm(true)
+  }
+  let attended = false
+  if(token){
+    const { id: userId } = jwt_decode(token) as {iat: number, id: number};
+    if(props.attendees){
+      attended = props.attendees.some( attendee => (attendee.id === userId))
```
Comment (sammyluo) [inline src/pages/EventDetailPage/Event.tsx:53]: Very nice! 😄

File: src/pages/EventDetailPage/EventDetailPage.tsx [hunk 4]
```diff
@@ -14,18 +20,36 @@ const withEvent = (ComPonent:typeof Event) => {
...
+    let reviews: EventReviewType[] = event.reviews ? event.reviews : []
+    reviews.push(review)
+  
+    setEvent({...event,  reviews})
+  } 
+
+  useTracing()
   useEffect(()=>{
-    request(process.env.REACT_APP_API_END_POINT + `events/${id}`, dataHandler)
+    const headers = {
```
Comment (sammyluo) [inline src/pages/EventDetailPage/EventDetailPage.tsx:39]: You can make a util function to generate this object.

File: src/pages/LoginPage/LoginPage.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  return (
+    <div className={styles['page']}>
+      <div className={styles['container']}>
+          <Typography.Title className={styles['title']}>Login</Typography.Title>
+          <Divider/>
+        <div className={styles['form_container']}>
+          <Form
+            {...formItemLayout}
+            form={form}
+            name="register"
```
Comment (sammyluo) [inline src/pages/LoginPage/LoginPage.tsx:75]: Should this be `login`?

File: src/pages/peoplePage/PeoplePage.tsx [hunk 6]
```diff
@@ -1,8 +1,98 @@
...
+  
+  const dataHandler = (data: HostType[]) => {
+    const numOfLines = data? Math.ceil(data.length/Number_Of_Cards) : Number_Of_Lines
+    setUsers(data) 
+    if(numOfLines> Number_Of_Lines){ 
+      setMaxLines(numOfLines)
+    }
+  }
+
+  const trackingScrolling = () => {
```
Comment (sammyluo) [inline src/pages/peoplePage/PeoplePage.tsx:39]: Very nice implementation! 😄 👍

**Review by sammyluo — APPROVED** [review]
Awesome work, Terry! 😄  I really like your infinite scroll solution! Very cool! 👍 

Just left some minor comments. Also, if a file doesn't contain any JSX, you could name the file with .ts instead of .tsx. Like your slice files.

**Pull request merged**